### ISSUE TITLE
[Bug]: Missing header cell in single select Table and DataGrid doc examples

### ISSUE THREAD

**Issue by adamsamec** [microsoft/fluentui#27968]
### Library

React Components / v9 (@fluentui/react-components)

### System Info

```shell
N/A
```

### Are you reporting Accessibility issue?

yes

### Reproduction

Reproducable in the current doc

## Actual Behavior
In the documentation for the Fluent UI v9 Table and DataGrid components, in the Single Select variant, there is a header cell missing for the first table column, that is for the column with the selection radios. Due to this missing first column header, there is a mismatch for the respective header columns that are read by the screen reader as headers when navigating the table from one column to another.

## Expected Behavior
The first column header cell should be implemented as a static text labelled using the aria-label attribute, contained in a &lt;th&gt; element.

### Requested priority

Blocking

### Are you willing to submit a PR to fix?

no

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.
